src/main/java/org/jabref/gui/maintable/MainTableColumn.java

```diff
@@ -1,15 +1,35 @@
 package org.jabref.gui.maintable;
 
-import javafx.beans.value.ObservableValue;
+import java.util.Optional;
+
+import javafx.scene.Node;
+import javafx.scene.control.Label;
 import javafx.scene.control.TableColumn;
 
-abstract class MainTableColumn<T> extends TableColumn<BibEntryTableViewModel, T> {
+import org.jabref.gui.icon.JabRefIcon;
+
+public class MainTableColumn<T> extends TableColumn<BibEntryTableViewModel, T> {
 
-    MainTableColumn(String text) {
-        super(text);
+    private MainTableColumnModel model;
 
-        setCellValueFactory(param -> getColumnValue(param.getValue()));
+    private final Optional<JabRefIcon> iconLabel;
+
+    public MainTableColumn(MainTableColumnModel model) {
+        this.model = model;
+        this.iconLabel = Optional.empty();
     }
 
-    abstract ObservableValue<T> getColumnValue(BibEntryTableViewModel entry);
+    public MainTableColumnModel getModel() { return model; }
+
+    public Node getHeaderLabel() {
```

Comment:
I extracted this method out of the former NormalTableColumn, now FieldTableColumn, but it is not used. Should I remove it or keep it?

Reply:
If it's not used remove it